package.json

```
@@ -88,7 +88,11 @@
     "typescript": "^5.7.3"
   },
   "lint-staged": {
-    "*.{tsx?,mjs,sh,cjs,json,md,mdx,graphql,yml}": "prettier --write",
+    "*.{md,mdx}": [
+      "awk 'BEGIN {in_block=0} /^```/ { if (!in_block && /^```$/) exit 1; in_block = !in_block }'",
```

Something like this should give nicer error messages (allowing multiple errors per file and setting the exit code at the end if there were any):
```suggestion
      "awk 'BEGIN {in_block=0; err=0} /^```/ { if (!in_block && /^```$/) { printf("Missing language tag in code block on line %s\n", NR); err=1}; in_block = !in_block } END { exit err }",
```

`NR` is awk syntax for "record" number, which means line number in this case.